Source/ViewableController.swift

```diff
@@ -280,7 +289,17 @@ class ViewableController: UIViewController {
 
         switch viewable.type {
         case .image:
+            // Needed to avoid showing the loading indicator for a fraction of a second. Thanks to this the
+            // loading indicator will only be displayed when the image is taking a lot of time to load.
+            let deadline = DispatchTime.now() + Double(Int64(0.5 * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
+            DispatchQueue.main.asyncAfter(deadline: deadline) {
+                if self.imageView.image == nil {
+                    self.imageLoadingIndicator.startAnimating()
+                }
+            }
+
             viewable.media { image, _ in
+                self.imageLoadingIndicator.stopAnimating()
```

Comment:
Am I correct in my assumption that this will always get called back on the main thread?

Reply:
Will be called in the main thread :)

Reply:
This is a protocol that any Viewable should implement, so it's the responsability of the called to return this in the main thread. I could always add an extra check, though.